messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/bridge/JMSBridgeHandler.java

```diff
@@ -47,101 +45,110 @@
+            if (!readOnly) {
+                throw ControllerLogger.ROOT_LOGGER.managementResourceNotFound(context.getCurrentAddress());
             }
+            return;
         }
-        else if (START.equals(operationName)) {
-            try {
-                // we do not start the bridge directly but call startBridge() instead
-                // to ensure the class loader will be able to load any external resources
-                JMSBridgeService service = (JMSBridgeService) bridgeService.getService();
-                service.startBridge();
-            } catch (Exception e) {
-                context.getFailureDescription().set(e.getLocalizedMessage());
-            }
-        } else if (STOP.equals(operationName)) {
-            try {
-                bridge.stop();
-            } catch (Exception e) {
-                context.getFailureDescription().set(e.getLocalizedMessage());
-            }
-        } else if (PAUSE.equals(operationName)) {
-            try {
-                bridge.pause();
-            } catch (Exception e) {
-                context.getFailureDescription().set(e.getLocalizedMessage());
-            }
-        } else if (RESUME.equals(operationName)) {
-            try {
-                bridge.resume();
-            } catch (Exception e) {
-                context.getFailureDescription().set(e.getLocalizedMessage());
-            }
-        } else {
-            throw MessagingLogger.ROOT_LOGGER.unsupportedOperation(operationName);
+        final JMSBridge bridge = JMSBridge.class.cast(bridgeService.getValue());
+        switch (operationName) {
+            case READ_ATTRIBUTE_OPERATION:
+                readAttributeValidator.validate(operation);
+                final String name = operation.require(NAME).asString();
+                if (STARTED.equals(name)) {
+                    context.getResult().set(bridge.isStarted());
+                } else if (PAUSED.getName().equals(name)) {
+                    context.getResult().set(bridge.isPaused());
+                }  else if (CommonAttributes.MESSAGE_COUNT.getName().equals(name)) {
+                    context.getResult().set(bridge.getMessageCount());
+                }  else if (ABORTED_MESSAGE_COUNT.getName().equals(name)) {
+                    context.getResult().set(bridge.getAbortedMessageCount());
+                } else {
+                    throw MessagingLogger.ROOT_LOGGER.unsupportedAttribute(name);
+                }
+                break;
+            case START:
+                try {
+                    // we do not start the bridge directly but call startBridge() instead
+                    // to ensure the class loader will be able to load any external resources
+                    JMSBridgeService service = (JMSBridgeService) bridgeService.getService();
+                    service.startBridge();
+                } catch (Exception e) {
+                    context.getFailureDescription().set(e.getLocalizedMessage());
```

Comment:
This applies to all these cases, and is an existing problem...

These all involve failed manipulation of the server runtime state but this handling means nothing about them will appear in the server log.  Can these just throw on any RuntimeException, wrap any checked exception in RuntimeException and throw on, and count on the throwable handling in AbstractOperationContext.executeStep (e.g. currently https://github.com/wildfly/wildfly-core/blob/master/controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java#L1032) to log and prepare a response to the client?  Even the failure-description it prepares at L1037 is likely to be more meaningful than e.getLocalizedMessage() which typically doesn't include the exception class and often isn't understandable without that.